internal/service/organizations.go

```diff
@@ -151,3 +198,132 @@ func (s *DashNGoImpl) UploadOrganizations() []string {
 	}
 	return result
 }
+
+// SwitchOrganization switch organization context
+func (s *DashNGoImpl) SwitchOrganization(id int64) error {
+	if !s.grafanaConf.IsBasicAuth() {
+		log.Warnf("Basic auth required for Org switching.  Ignoring Org setting and continuing")
+		return nil
+	}
+	valid := false
+	if id > 1 {
+		var orgsPayload []*models.OrgDTO
+		orgList, err := s.client.Orgs.SearchOrgs(orgs.NewSearchOrgsParams(), s.getBasicAuth())
+		if err != nil {
+			log.Warn("Error fetch organizations requires (SuperAdmin Basic Auth), assuming valid ID was requested.  Cannot validate OrgId")
```

Comment:
Is there any reason not to bail here as well? This seems like an invalid config / setup, what's the use case in continuing forward?

Reply:
That would then require SuperAdmin for all Org namespaces orgs which is almost every call, that would make GDG unusable for too many people.  

For basic auth, we assume the value is correct and attempt to switch context, if the value is invalid, it will fail later in the call stack.  If we have a superadmin, then we validate the the org exists.   I could remove the entire pattern and just always assume, but I think the error is prettier and more helpful if it's a 'Org is missing' vs some call failed.